File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n1 = word1.size();
        int n2 = word2.size();
        
        string s = "";

        int i=0, j=0;
        
        while(i<n1 && j<n2)
        {
            //alternative insertion
            s += word1[i++];
            s += word2[j++];
        }    
        
        //edge cases
        //if the word exceeds
        
        while(i<n1)
            s += word1[i++];
        
        while(j<n2)
            s += word2[j++];
        
        
        return s;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        //it is similar to c++ string = ""
        StringBuilder result = new StringBuilder();
        
        int iter = 0;
        //will iterate one by one in each word
        while(iter < word1.length() || iter < word2.length()){
            if(iter < word1.length()){
                result.append(word1.charAt(iter));
            }
            if(iter < word2.length()){
                result.append(word2.charAt(iter));
            }
                
            iter++;
        }
        return result.toString();
    }
}